test/ServiceBus.Tests/PlugableQueueBalancerTests/LeaseManagerGrain.cs

```diff
@@ -0,0 +1,106 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+using Orleans;
+using Orleans.Runtime;
+using Orleans.Runtime.Configuration;
+using Orleans.Streams;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Tester.StreamingTests
+{
+    //one lease manager grain per stream provider, so its key is stream provider name
+    public interface ILeaseManagerGrain : IGrainWithStringKey
+    {
+        Task<QueueId> Acquire();
+        Task<bool> Renew(QueueId leaseNumber);
+        Task Release(QueueId leaseNumber);
+        Task<int> GetLeaseResposibility();
+
+        //methods used in test asserts
+        Task RecordBalancerResponsibility(string balancerId, int ownedQueues);
+        Task<Dictionary<string, int>> GetResponsibilityMap();
+
+    }
+
+    public class LeaseManagerGrain : Grain, ILeaseManagerGrain
+    {
+        //queueId is the lease id here
+        private static DateTime UnAssignedLeaseTime = DateTime.MinValue;
+        private Dictionary<QueueId, DateTime> queueLeaseToRenewTimeMap;
+        private ISiloStatusOracle siloStatusOracle;
+        private IStreamQueueMapper queueMapper;
+        private ClusterConfiguration clusterConfiguration;
+        public override Task OnActivateAsync()
+        {
+            this.siloStatusOracle = base.ServiceProvider.GetRequiredService<ISiloStatusOracle>();
+            this.clusterConfiguration = base.ServiceProvider.GetRequiredService<ClusterConfiguration>();
+            this.queueMapper = base.ServiceProvider.GetServiceByName<IStreamQueueMapper>(this.GetPrimaryKeyString());
```

Comment:
"LeaseManager being grain" has been discussed as a possibility. I want to make sure LeaseManager**Grain**BasedQueueBalancer can be an option later. 

Just want to bring up the concern that, if we go with "Lease manager being a grain" option, assign it complex dependency will not be intuitive with this PR's interface design. 

As an example: I found it difficult for LeaseManagerGrain to access StreamQueueMapper, which intuitively should be its dependency. This is a pretty hacky way to access it. We can think of other more elegant way in the future.